File: Array&String/Sorting/can_make_arithmetic_progression_from_sequence.py
class Solution:
    def canMakeArithmeticProgression(self, arr: list[int]) -> bool:
        arr = sorted(arr)
        diff = arr[1] - arr[0]
        n = len(arr)
        if n == 2: return True
        for i in range(2, n):
            if arr[i] - arr[i-1] != diff: return False
        return True

File: Array&String/Sorting/CPP/can_make_arithmetic_progression_from_sequence.cpp
#include <algorithm>
#include <vector>
using namespace std;

class Solution {
    public:
        bool canMakeArithmeticProgression(vector<int>& arr) {
            sort(arr.begin(), arr.end());
            int diff = arr[1] - arr[0], n = arr.size();
            if (n == 2) return true;
            for (int i = 2; i < n; i++){
                if (arr[i] - arr[i-1] != diff) return false;
            }
            return true;
        }
    };